c++/include/orc/Vector.hh

```diff
@@ -194,6 +194,14 @@ namespace orc {
     int32_t scale;
   };
 
+  /**
+   * Compares two decimals
+   * @param decimal1 1st decimal to compare
+   * @param decimal2 2nd decimal to compare
+   * @return true is decimal1 is less than decimal2; otherwise false
+   */
+  bool operator<(Decimal decimal1, Decimal decimal2);
```

Comment:
Libraries usually don't define these operators in the public API.
It is also odd that we only define one operator and not the other operators.
All other non-primitive types don't have these operators defined as well. It looks odd just to add this one. Can we just use `compare(Decimal, Decimal)` api inside?